App.js

```diff
@@ -306,25 +313,41 @@ export default class App extends React.Component {
             linking={linking}
             theme={navTheme}
             onStateChange={(state) => {
-              console.log('STATE CHANGED: ', state);
+              const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
+              const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
+              const isRHPOnTopOfStack = lastRHPIndex === state.index;
+              this.stateRef.current = {
+                ...this.stateRef.current,
+                state,
+                lastChatIndex,
+                lastRHPIndex,
+                isRHPOnTopOfStack
+              }
             }}
             initialState={this.state.initialState}
           >
             <Stack.Navigator
               initialRouteName="LeftHandNav"
+              screenOptions={{headersShown: false}}
             >
                 <Stack.Screen
                   name="LeftHandNav"
                   component={LeftHandNav}
-                  options={{headerShown: false}}
+                  options={{isLHNVisible: (this.stateRef.state.index === 0 && this.stateRef.routes.length === 1) || !isSmallScreenWidth}}
```

Comment:
I am kind of confused about why we are passing visibility information here. I _think_ the only option we'd pass here is just something like: `isSidebar: true` and then have the navigator deal with that if it needs to.

Reply:
Looks like a good suggestion, let me try it.